demo/api/schema.gql

```diff
@@ -356,6 +362,7 @@ type Query {
   pageTreeNode(id: ID!): PageTreeNode
   pageTreeNodeByPath(path: String!, scope: PageTreeNodeScopeInput!): PageTreeNode
   pageTreeNodeList(scope: PageTreeNodeScopeInput!, category: String): [PageTreeNode!]!
+  paginatedPageTreeNodes(scope: PageTreeNodeScopeInput!, category: String, sort: [PageTreeNodeSort!], offset: Int = 0, limit: Int = 20): PaginatedPageTreeNodes!
```

Comment:
(incompatible) alternative would be to make pageTreeNodeList (optionally) pageable

Reply:
What would happen when no paging parameters are set? Return all page tree nodes?

Reply:
Currently, the default parameters of offsetBasedPagination are used.

Reply:
What should happen if no paging parameters are set? should the `pageTreeNodeList` query still return all page tree nodes, or should it use the default paging parameters (offset 0, limit 20). I don't believe we should go for the second option, as it is a massive breaking change in all our applications, and requires additional changes in the page tree module. The first option seems odd, but may work. 

For the redirects, we decided to not add optional paging, but add a second query instead (and remove the non-pageable query in the future).